GarnerCorp/build-actions
# Prettier

The prettier action introduced in #40 was unfortunately an untested stub.

These commits make it a real working action.

Unlike the other PRs where I'm using rebase-merge, this PR will be a squash merge as the individual commits aren't interesting to someone has previously used the action (as no one could use the action, it had syntax errors).

You can see this action doing work here:
https://github.com/check-spelling-sandbox/build-actions-prettier-0/pull/1


```yaml
name: Prettier

on: push

jobs:
  prettier:
    name: Prettier
    runs-on: ubuntu-latest
    permissions:
      contents: write

    steps:
      - name: Prettier
        uses: garnercorp/build-actions/prettier@updates
        with:
          prettier-options: "--write"
          push-to-remote: true
          update-git_blame_ignore_revs: true
          checkout-ssh-key: "${{ secrets.CHECK_SPELLING }}"
```

`inputs.checkout-ssh-key` is optional, but will supersede the `github.token` from `contents: write`. If you already have a token (like `secrets.CHECK_SPELLING`) then you don't need to include `contents: write` and could instead use `permissions: {}`.

This specific configuration (with both `contents: write` and `checkout-ssh-key`) is designed for dropping into repositories at the same time as spelling.yml under the assumption that `spelling.yml` will _eventually_ walk an admin through the steps to create `secrets.CHECK_SPELLING` -- as happened in https://github.com/check-spelling-sandbox/build-actions-prettier-0/pull/1#issuecomment-1930794986.

The one thing I haven't spent time investigating is whether it's possible for this workflow to:
1. run
2. call prettier which generates changes
3. create a commit
4. push that commit (using `checkout-ssh-key` or similar)
5. which in turn triggers the workflow to run again
6. call prettier which generates changes that don't match what was committed in 3.

If this could happen, the workflow would trigger an infinite loop, which would be pretty bad.

I don't think it should, since the output from prettier should be stable.
